persistence/nosql/async/api/src/main/java/org/apache/polaris/async/Cancelable.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.async;
+
+import java.util.concurrent.CompletionStage;
+
+public interface Cancelable<R> {
```

Comment:
Can we have some javadoc comments explaining the difference between this and `Future` and why we want to partially hide the Futures returned by executors under this interface?

Can callers still get a future via `completionStage().toCompletableFuture()`? Why is it a "has a future" pattern instead of an "is a future" pattern?